src/lib/utils/error-logger.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const logger = {};
+
+const getLogData = (err) => {
+    let logData;
+    if (err.stack) {
+        logData = err.stack;
+    } else if (err.message) {
+        logData = err.message;
+    } else if (err.toString instanceof Function) {
+        logData = err.toString();
+    } else {
+        logData = err;
+    }
+
+    return logData;
+};
+
+module.exports.register = (server, options, next) => {
+
+    //logs error
+    logger.logError = (tags,err) => {
+        server.log(['error'].concat(tags), getLogData(err));
+    };
+
+    //logs and throws error
+    logger.logAndThrowError = (tags,err) => {
```

Comment:
Cool stuff.

> nodeapi application uses Error objects to transmit information ... as evidenced by their use of Boom

Yea, I probably assumed too much, there!  There's no reason to not wrap a client Error-ish thing in there.

> they can be used for flow control

It's a new concept that feels a little uncomfortable.  I'm not saying I'm opposed to it, but it does raise some questions.  Traditionally, these tools are used only when something goes _wrong_.  Now, we are using them when things also go _right_.  I can see how this would be beneficial to exit the route chain quickly when authentication has received bad un/pw.  Maybe my attitude is actually bogus here too.  Maybe the fact that a `401` happened _is a non-standard expectation_, so the server is in an error state, and it happens to be one that knows how to recover from!  OK ok ok, ya ya ya.  :bulb: 

Got it!  I'm on board!  Thanks man.